### PR TITLE
feat: migrate from Monaco to CodeMirror

### PR DESCRIPTION
**Pull request by olexii4**
### What does this PR do?
Migrate from Monaco to CodeMirror.

### What issues does this PR fix or reference?
fixes https://github.com/eclipse/che/issues/22481, https://github.com/eclipse/che/issues/22237

### Is it tested? How?

Before migration:
Знімок екрана 2023-10-09 о 12 30 08

After migration:
Знімок екрана 2023-10-09 о 12 15 22

Знімок екрана 2023-10-09 о 06 38 21
Знімок екрана 2023-10-09 о 06 38 29

### CODE REVIEW

**Review by ibuziuk — APPROVED** [review]

**Comment by ibuziuk** [comment]
@olexii4 LGTM, could you clarify how the devfile schema is going to be validated now? Asking in the context of https://github.com/eclipse/che/issues/21845 ?

**Comment by olexii4** [comment]
> @olexii4 LGTM, could you clarify how the devfile schema is going to be validated now? Asking in the context of [eclipse/che#21845](https://github.com/eclipse/che/issues/21845) ?

@ibuziuk It is still validating on `che-server` side during factory resolving.
Now we don't have the ability to change a devfile from the dashboard client side. This schema API became useless. 

Previously we used to use these schemas to validate a devfile with the editor language server when we tried to change a devfile.

File: packages/dashboard-frontend/package.json [hunk 1]
```diff
@@ -38,6 +38,7 @@
     "@patternfly/react-table": "^4.113.3",
     "axios": "^0.21.4",
     "buffer": "^6.0.3",
+    "codemirror": "^5.65.15",
```
[the remark this answers is not in the document]
Comment (olexii4) [reply packages/dashboard-frontend/package.json:41]: I have fixed it.

File: packages/dashboard-frontend/src/store/DevfileRegistries/index.ts [hunk 2]
```diff
@@ -268,40 +266,6 @@ export const actionCreators: ActionCreators = {
       }
     },
 
-  requestJsonSchema:
```
[the remark this answers is not in the document]
Comment (olexii4) [reply packages/dashboard-frontend/src/store/DevfileRegistries/index.ts:271]: I have fixed it.

**Comment by vinokurig** [comment]
>@ibuziuk It is still validating on che-server side during factory resolving.
>Now we don't have the ability to change a devfile from the dashboard client side. This schema API became useless.
>
>Previously we used to use these schemas to validate a devfile with the editor language server when we tried to change a devfile.

If we are going to return back the devfile validation mechanism to the dashboard side, I am +1 to remove the validation check from the che-server side.

**Review by akurinnoy — APPROVED** [review]

**Pull request merged**